confbot_driver/include/confbot_driver/confbot_driver.hpp

```diff
@@ -87,22 +113,88 @@ class ConfbotDriver : public rclcpp::Node
     tf_broadcaster_->sendTransform(msg_);
   }
 
-  void update_position(std::shared_ptr<geometry_msgs::msg::Twist> twist_msg)
+  void update_velocity(std::shared_ptr<geometry_msgs::msg::Twist> twist_msg)
```

Comment:
What is the behavior if a goal is running and messages are received on the `cmd_vel` topic ?
The speed specified in the action goal is ignored and the one from `cmd_vel` is used instead?

Reply:
Did this change since last review? if not, can you please clarify the behavior here?

Reply:
Behavior did not change. I thought we had clarify this offline. 
The behavior is exactly as described. When a second twist message is sent during the execution of the action, the new velocity is applied.

Reply:
Oh I guess I misunderstood, my point was that it doesn't seem to be expected behavior. 
Changing the goal but continuing the action will be confusing on both sides:
- for the person sending on `cmd_vel` (at the end of the action): Why did the robot stop suddenly?
- for the action client: "How did my robot travel so far when I asked it to go only to 0.1?"

IMO the message on `cmd_vel` should either be ignored, or overrule the action and make the action fail.

Reply:
I wanted to keep the code as simple as possible for the demo. I totally see your point, but you'll run into similar things if you spawn a second twist publisher or issue a action client goal twice concurrently. 

in 21425e3 I implemented that the action client always has a higher priority than the subscriber. In the end this also makes the demo easier because no matter what speed the twist publisher currently publishes, the action client always comes through.

Reply:
Sounds good.:+1:

>  issue a action client goal twice concurrently.

I didn't look at ROS2 actions in details, but if there's a way for the goals to be queued and executed sequentially by the server, I believe it would be better. But it's not a blocker at all